packages/d/dotnet/package.yml

```diff
@@ -0,0 +1,82 @@
+networking : yes
+libsplit   : no
+clang      : no # current clang version breaks the build. We can try again when llvm is updated
+strip      : no
+builddeps  :
+    - pkgconfig(icu-i18n)
+    - pkgconfig(krb5)
+    - pkgconfig(libcurl)
+    - pkgconfig(libssl)
+    - pkgconfig(libunwind)
+    - pkgconfig(libxml-2.0)
+    - pkgconfig(lttng-ust)
+    - pkgconfig(zlib)
+    # - dotnet-sdk
+    # - dotnet-source-built-artifacts
+    - git
+    - lldb-devel
+    - ninja
+rundeps    :
+    - kerberos-devel
+    - libicu
+    - liblttng-ust
+    - libunwind-devel
+    - openssl-devel
+    - sdk:
+        - dotnet
+    - zlib-devel
+setup      : |
+    %patch -p1 -i $pkgfiles/1005-default-opt-out-of-telemetry.patch
+
+build      : |
+    # EXTRA_* flags are passed to the build system
+    export EXTRA_CFLAGS="$CFLAGS"
+    export EXTRA_CXXFLAGS="$CXXFLAGS"
+    export EXTRA_LDFLAGS="$LDFLAGS"
+    if [ -d /usr/lib/dotnet ]; then
+        cp -r /usr/lib/dotnet .dotnet
+    fi
+    if [ -d /usr/share/dotnet/source-built-artifacts ]; then
+        ln -sf /usr/share/dotnet/source-built-artifacts/Private.SourceBuilt.Artifacts.*.tar.gz prereqs/packages/archive/
+        ln -sf /usr/share/dotnet/source-built-artifacts/Private.SourceBuilt.Prebuilts.*.tar.gz prereqs/packages/archive/
+    fi
+    # setting `version` (any case) clobbers the msbuild version target
+    unset CFLAGS CXXFLAGS LDFLAGS version
+    ./prep.sh
+    ./build.sh --clean-while-building --online
+install    : |
+    # required bootstrap artifacts
+    install -Dm00644 artifacts/x64/Release/Private.SourceBuilt.Artifacts.*.tar.gz -t $installdir/usr/share/dotnet/source-built-artifacts/
+    install -Dm00644 artifacts/x64/Release/Private.SourceBuilt.Prebuilts.*.tar.gz -t $installdir/usr/share/dotnet/source-built-artifacts/
+
+    # sdk
+    install -dm00755 $installdir/usr/{bin,lib/dotnet,share/licenses/dotnet}
+    tar -xf artifacts/x64/Release/dotnet-sdk-*.tar.gz -C $installdir/usr/lib/dotnet
+    mv $installdir/usr/lib/dotnet/{LICENSE.txt,ThirdPartyNotices.txt} $installdir/usr/share/licenses/dotnet/
+    ln -s /usr/lib/dotnet/dotnet $installdir/usr/bin/dotnet
+    ln -s /usr/lib/dotnet/host/fxr/$version/libhostfxr.so $installdir/usr/lib/libhostfxr.so
+
+    # completions
+    install -Dm00644 src/sdk/scripts/register-completions.bash $installdir/usr/share/bash-completion/completions/dotnet
+    install -Dm00644 src/sdk/scripts/register-completions.zsh $installdir/usr/share/zsh/site-functions/_dotnet
+    # install -Dm00644 src/sdk/scripts/register-completions.ps1 $installdir/usr/local/share/powershell/Modules
+patterns   :
+    - sdk:
+        - /usr/lib/dotnet/metadata
+        - /usr/lib/dotnet/packs
+        - /usr/lib/dotnet/sdk
+        - /usr/lib/dotnet/sdk-manifests
+        - /usr/lib/dotnet/templates
+    - source-built-artifacts:
+        - /usr/share/dotnet/source-built-artifacts
```

Comment:
If you're deprecating fsharp you should probably have `replaces` keys to make sure that dotnet is installed automatically in it's place.